yt_dlp/extractor/kaltura.py

```diff
@@ -233,10 +234,13 @@ def _extract_embed_urls(cls, url, webpage):
             embed_url = 'kaltura:%(partner_id)s:%(id)s' % embed_info
             escaped_pid = re.escape(embed_info['partner_id'])
             service_mobj = re.search(
-                r'<script[^>]+src=(["\'])(?P<id>(?:https?:)?//(?:(?!\1).)+)/p/%s/sp/%s00/embedIframeJs' % (escaped_pid, escaped_pid),
+                r'<script[^>]+src=(["\'])(?P<id>(?:https?:)?//(?:(?!\1).)+)/p/%s/sp/%s00/embedIframeJs/uiconf_id/(?P<uiconf_id>[^/]+)' % (escaped_pid, escaped_pid),
                 webpage)
             if service_mobj:
-                embed_url = smuggle_url(embed_url, {'service_url': service_mobj.group('id')})
+                embed_url = smuggle_url(embed_url, {'service_url': service_mobj.group('id'), 'uiconf_id': service_mobj.group('uiconf_id')})
+            flashvars_mobj = re.search(r'flashvars\s*=\s*(?P<flashvars>.+);\n', webpage)
```

Comment:
`.+` may match too much. Isn't it a json? If so, use `_search_json` instead for more robust mathing

Reply:
What's the best way to call that from a class method?

Reply:
hm.. I forgot it's a classmethod. Calling it from here would need some refactoring. Leave it be for now